### PR TITLE
Check received messages

### PR DESCRIPTION
**Pull request by denisraslov**
http://cultofmartians.com/tasks/logux-messages-check.html

### CODE REVIEW

**Comment by ai** [comment]
Looks good. But right now we have too many validation code in `messages/*`.  Could you create some `validator` and move checking and error creation there? `messages/*` should be about sync logic, not validation.

**Comment by denisraslov** [comment]
Sure, I agree, will do it soon.

File: messages/connect.js [hunk 1]
```diff
@@ -120,21 +111,11 @@ module.exports = {
...
-  validateConnected: function validateConnected (ver, nodeId, time, options) {
-    return TypeChecker.checkType(nodeId, 'string', true) &&
-      TypeChecker.checkType(time, 'array', true) && time.length === 2 &&
-      TypeChecker.checkType(time[0], 'number', true) &&
-      TypeChecker.checkType(time[1], 'number', true) &&
-      TypeChecker.checkType(options, 'object', false)
-  },
-
   connectedMessage: function connectedMessage (ver, nodeId, time, options) {
     var isValid = this.validateConnected(ver, nodeId, time, options)
```
Comment (ai) [inline messages/connect.js]: Why you need `isValid` here? Throw a error from `validateConnected` ;)

File: validator.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+var TypeChecker = require('./type-checker')
```
Comment (ai) [inline validator.js]: You don’t use `TypeChecker` anywhere else. Move it to `validator.js`. It is a simple system, we should not have too many validation files in project.

File: messages/ping.js [hunk 3]
```diff
@@ -7,11 +7,21 @@ module.exports = {
...
+      this.wrongFormatError(['ping', synced])
+      return
+    }
+
     this.setOtherSynced(synced)
     this.send(['pong', this.log.lastAdded])
   },
 
   pongMessage: function pongMessage (synced) {
+    if (!this.validatePong(synced)) {
```
Comment (ai) [inline messages/ping.js]: Move all `validatePong`, etc methods to `validateMessage`. Just use `message[0]` to detect message type and validate it there.

**Comment by denisraslov** [comment]
Done. Any other comments? :)
Now it seems that we need to move tests for message validation to `validator.test.js`. Am I right?

**Comment by ai** [comment]
Good question :-/. I will think tomorrow.

Anyway right now it looks pretty good. Well done :+1:

**Comment by ai** [comment]
Right now you could fix git conflicts from my today changes :D

**Comment by denisraslov** [comment]
Cool 😎
As I see, you've done it already :)

**Comment by ai** [comment]
I am still in merging :(

**Pull request merged**